src/Adapter/MSTest.CoreAdapter/Helpers/ReflectHelper.cs

```diff
@@ -130,6 +130,46 @@ public bool HasAttributeDerivedFrom(MemberInfo memberInfo, Type baseAttributeTyp
             return false;
         }
 
+        /// <summary>
+        /// Returns true when specified class/member has attribute which implements specific interface.
+        /// </summary>
+        /// <param name="memberInfo">The member info.</param>
+        /// <param name="interfaceType">The interface type.</param>
+        /// <param name="inherit">Should look at inheritance tree.</param>
+        /// <returns>Returns bool specifying whether class/member has attribute which implements given interface.</returns>
+        public virtual bool HasAttributeImplementingInterface(MemberInfo memberInfo, Type interfaceType, bool inherit)
+        {
+            if (memberInfo == null)
+            {
+                throw new ArgumentNullException(nameof(memberInfo));
+            }
+
+            if (interfaceType == null)
+            {
+                throw new ArgumentNullException(nameof(interfaceType));
+            }
+
+            // Get all attributes on the member.
+            Dictionary<string, object> attributes = this.GetAttributes(memberInfo, inherit);
+
+            if (attributes != null)
+            {
+                // Try to find the attribute that implements interfaceType.
+                foreach (object attribute in attributes.Values)
+                {
+                    Debug.Assert(attribute != null, "ReflectHelper.HasAttributeImplementingInterface: internal error: wrong value in the attrs dictionary.");
```

Comment:
how does Debug.Assert help? also if we hit this condition we will fail any ways in the next line